### PR TITLE
[BUGFIX LTS] Don't run getters while applying mixins

### PR DESCRIPTION
**Pull request by wycats**
This change ensures that getters are never evaluated while applying
mixins.

It relies on the fact that all getters (including undecorated ones) get
converted into classic decorators when the mixin is originally created.

### CODE REVIEW

**Comment by chriskrycho** [comment]
Looks like this is aimed at the issue ID'd in #19916?

**Comment by gitKrystan** [comment]
@chriskrycho It's related to https://github.com/emberjs/ember.js/issues/18860 and https://github.com/emberjs/ember.js/pull/20129

**Comment by chriskrycho** [comment]
Yep, and #20129 was a failing test for #19916. 👍🏼

**Comment by wycats** [comment]
@chriskrycho Do you mean to refer to "No documented public API for `Owner`"?

**Comment by chriskrycho** [comment]
Nope, brain fog. I meant #18860, which Krystan linked. 😮‍💨

File: tsconfig.json [hunk 1]
```diff
@@ -4,6 +4,6 @@
     "outDir": "dist",
     "noEmit": true
   },
-  "include": ["packages/**/*.ts"],
+  "include": ["packages/**/*.ts", "packages/**/*.js"],
```
Comment (chriskrycho) [inline tsconfig.json:7]: Hmm, why this change? We're intentionally *not* including `.js` files here, and also don't have `checkJs: true`, so it shouldn't particularly affect the behavior of the compiler. 🤔

File: packages/@ember/object/mixin.ts [hunk 2]
```diff
@@ -8,12 +8,14 @@ import { guidFor, observerListenerMetaFor, ROOT, wrap } from '@ember/-internals/
...
-  ComputedPropertyObj,
-  ComputedPropertySetter,
-  ComputedDescriptor,
+import {
+  type ComputedDecorator,
+  type ComputedPropertyGetter,
+  type ComputedPropertyObj,
+  type ComputedPropertySetter,
+  type ComputedDescriptor,
+  isAccessor,
```
Comment (chriskrycho) [inline packages/@ember/object/mixin.ts:17]: You imported this but didn't use it *and* it doesn't appear to be used below, so I think we can simply drop this line from the `import` statement?

```suggestion
```

**Review by chriskrycho — CHANGES_REQUESTED** [review]
Couple small things and then we should be able to land and back port this. Thanks for getting it fixed!

File: tsconfig.json (hunk 1 above)
Comment (gitKrystan) [reply tsconfig.json:7]: @wycats Mentioned to me that this changed fixed some issues with auto-import in VSCode. If we decide to keep it, it should probably go in a separate PR.

File: packages/@ember/object/mixin.ts (hunk 2 above)
Comment (gitKrystan) [reply packages/@ember/object/mixin.ts:17]: Interestingly, the [files tab](https://github.com/emberjs/ember.js/pull/20388/files) of this PR is showing linter (tsc) errors related to this, but they didn't fail CI. Is that configuration intentional?

File: tsconfig.json (hunk 1 above)
Comment (wycats) [reply tsconfig.json:7]: Yeah, totally agree. I meant to separate it out but I think we forgot.

Comment (wycats) [reply tsconfig.json:7]: TL;DR if you don't include the JS files in the tsconfig then the language server doesn't do any cross-module resolving in the JS files, which:

- breaks auto-import
- causes types from other modules to not resolve, which creates a lot of inferred `any`s for no reason (and breaking intellisense)
- causes some types to resolve to the published Ember types (which are present in node_modules), which is pretty confusing.

I meant to check whether jsconfig would be good enough to resolve these issues. Since you're the TS expert here, would you prefer to add the `js` files to `tsconfig` (without `checkJS`), or try to separate them into another config file.

This could either mean `jsconfig` (if it works) or creating a second `tsconfig` and link to it from the main one with a project reference.

I'm open to anything that addresses the workflow issues, and I'm totally fine with separating it into a separate PR.

**Comment by gitKrystan** [comment]
@chriskrycho I think this is ready

File: tsconfig.json (hunk 1 above)
Comment (chriskrycho) [reply tsconfig.json]: Ah, so this specifically ends up affecting tests, which are largely unconverted? Makes sense. I'm totally fine with `js` files in the config; I just wanted to understand where they had come up since all our actual implementation is now in TS—but "tests" makes sense. We can land that right after this. Thanks for explaining!

**Review by chriskrycho — APPROVED** [review]

**Comment by chriskrycho** [comment]
I am planning to do an LTS release (so: back to 4.8) by the end of this week and will make sure this is included!
